lib/services/ios-debug-service.ts

```diff
@@ -0,0 +1,290 @@
+import options = require("./../common/options");
+import iOSProxyServices = require("./../common/mobile/ios/ios-proxy-services");
+import iOSDevice = require("./../common/mobile/ios/ios-device");
+import net = require("net");
+import path = require("path");
+
+module notification {
+	export function waitForDebug(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.WaitForDebugger";
+	}
+
+	export function attachRequest(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.AttachRequest";
+	}
+
+	export function appLaunching(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.AppLaunching";
+	}
+
+	export function readyForAttach(bundleId: string): string {
+		return bundleId + ":NativeScript.Debug.ReadyForAttach";
+	}
+}
+
+class IOSDebugService implements IDebugService {
+	constructor(private $platformService: IPlatformService,
+		private $iOSEmulatorServices: Mobile.IEmulatorPlatformServices,
+		private $devicesServices: Mobile.IDevicesServices,
+		private $platformsData: IPlatformsData,
+		private $projectData: IProjectData,
+		private $childProcess: IChildProcess,
+		private $logger: ILogger,
+		private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $injector: IInjector,
+		private $npm: INodePackageManager) {
+	}
+
+	get platform(): string {
+		return "ios";
+	}
+
+	public debug(): IFuture<void> {
+		return (() => {
+			if ((!options.debugBrk && !options.start) || (options.debugBrk && options.start)) {
+				this.$logger.error("Expected exactly one of the --debug-brk or --start options.");
```

Comment:
this.$errors.failWithoutHelp seems more appropraite here. So you can omit the return.